cumulus-client/src/main/java/mil/army/usace/hec/cumulus/client/controllers/DownloadsController.java

```diff
@@ -0,0 +1,80 @@
+package mil.army.usace.hec.cumulus.client.controllers;
+
+import static mil.army.usace.hec.cumulus.client.controllers.CumulusEndpointConstants.ACCEPT_HEADER_V1;
+
+import hec.army.usace.hec.cumulus.http.client.AuthenticatedHttpRequestBuilderImpl;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.concurrent.CompletableFuture;
+import mil.army.usace.hec.cumulus.client.model.CumulusObjectMapper;
+import mil.army.usace.hec.cumulus.client.model.Download;
+import mil.army.usace.hec.cumulus.client.model.DownloadRequest;
+import mil.army.usace.hec.cwms.http.client.ApiConnectionInfo;
+import mil.army.usace.hec.cwms.http.client.HttpRequestBuilderImpl;
+import mil.army.usace.hec.cwms.http.client.HttpRequestResponse;
+
+public class DownloadsController {
+
+    private static final String DOWNLOADS_ENDPOINT = "downloads";
+
+    /**
+     * Retrieve Download (Used for obtaining download status).
+     *
+     * @param apiConnectionInfo    - connection info
+     * @param downloadsEndpointInput - download endpoint input containing download id
+     * @return Download
+     * @throws IOException - thrown if retrieve failed
+     */
+    public Download retrieveDownload(ApiConnectionInfo apiConnectionInfo, DownloadsEndpointInput downloadsEndpointInput)
+        throws IOException {
+        HttpRequestResponse response =
+            new HttpRequestBuilderImpl(apiConnectionInfo, DOWNLOADS_ENDPOINT + "/" + downloadsEndpointInput.getDownloadId())
+                .get()
+                .withMediaType(ACCEPT_HEADER_V1)
+                .execute();
+        return CumulusObjectMapper.mapJsonToObject(response.getBody(), Download.class);
+    }
+
+    /**
+     * Create a Download request.
+     * @param apiConnectionInfo - connection info
+     * @param downloadRequest - Download Request object containing start, end, watershed ID, and product IDs
+     * @return Download object containing URL to DSS File
+     * @throws IOException - thrown if POST request fails
+     */
+    public Download createDownload(ApiConnectionInfo apiConnectionInfo, DownloadRequest downloadRequest, String token)
```

Comment:
I updated to pass in a listener specific for dss file creation. This then spins up an asynch job that does the POST request and tracks progress of the dss file creation, and returns a CompleteableFuture<Download>, so the thread calling this can handle it as necessary.